src/com/owncloud/android/utils/FileStorageUtils.java

```diff
@@ -184,15 +184,7 @@ public static RemoteFile fillRemoteFile(OCFile ocFile){
         
         Collections.sort(files, new Comparator<OCFile>() {
             public int compare(OCFile o1, OCFile o2) {
-                if (o1.isFolder() && o2.isFolder()) {
-                    Long obj1 = o1.getModificationTimestamp();
-                    return val * obj1.compareTo(o2.getModificationTimestamp());
-                }
-                else if (o1.isFolder()) {
-                    return -1;
-                } else if (o2.isFolder()) {
-                    return 1;
-                } else if (o1.getModificationTimestamp() == 0 || o2.getModificationTimestamp() == 0){
+                if (o1.getModificationTimestamp() == 0 || o2.getModificationTimestamp() == 0){
```

Comment:
Yeah, but this is already the current behaviour.
So, we should discuss this in another issue?
(Although I do not know how often this problem might occur as the modification timestamp should always be set, or?